francoismassart/eslint-plugin-tailwindcss
# feat: support tailwind config in typescript

# Support tailwind config written in typescript as well as other common extensions

## Description

- Replace simple `require` on tailwind config with the `loadConfig` function from _tailwindcss_ package that supports typescript parsing
- Switch from hardcoded name for default config to `resolveDefaultConfigPath` function from _tailwindcss_ package that looks for one of `./tailwind.config.{js,cjs,mjs,ts}`
- Update _tailwindcss_ to v3.3.x in order to import the new functionality from the package
- Because of update beyond v3.3.0 an additional fixes were made for classnames-order rule tests since[ tailwindcss changed the ordering function at v3.3.0](https://github.com/tailwindlabs/tailwindcss/pull/10672).

Fixes #228

## Type of change


## How Has This Been Tested?

In a any project with _tailwindcss_ and _eslint-plugin-tailwindcss_ that utilises a JS-based config:
- define some custom theme in the config (I used `theme: {colors: {blueish: `#0000aa`}}`;
- use a class based on the custom theme (`text-blueish` in my case);
- ensure the rule does not work (`no-custom-classname` in my case);
- rename config to `ts` extension;
- add typings so that the file is not parsable by a JS engine;
- check that the rule reports the custom class (evidently because config is not loaded);
- switch the _eslint-plugin-tailwindcss_ in `package.json` to the local repo checked out from this branch;
- now the custom config shall be loaded correctly and the rule again does not report.

## Checklist:
